pom.xml

```diff
@@ -360,6 +392,84 @@
 					</execution>
 				</executions>
 			</plugin>
+
+
+			<plugin>
+				<groupId>org.jetbrains.kotlin</groupId>
+				<artifactId>kotlin-maven-plugin</artifactId>
+				<version>${kotlin.version}</version>
+				<configuration>
+					<args>
+						<arg>-Xjsr305=strict -Xopt-in=kotlin.RequiresOptIn</arg>
+					</args>
+					<compilerPlugins>
+						<plugin>spring</plugin>
```

Comment:
The `kotlin-maven-plugin` compiled the Kotlin files. The `spring/kotlin-maven-allopen` sub-plugin makes Spring annotated classes (`@Controller`, `@Service`, etc) `open` (non-final) which is needed by Spring for proxying/APO, etc. This is needed since Kotlin makes classes `final` by default.

Reply:
Valuable information.
Can we put that information into an XML comment (esp. those about `kotlin-maven-allopen`) here? Otherwise I fear it could get lost over time.